src/synapse/azext_synapse/custom.py

```diff
@@ -168,12 +181,30 @@ def delete_spark_pool(cmd, client, resource_group_name, workspace_name, spark_po
     return sdk_no_wait(no_wait, client.delete, resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, location, sku_name,
-                    sku_tier=None, max_size_bytes=None, source_database_id=None, recoverable_database_id=None,
-                    create_mode=None, tags=None, no_wait=False):
+def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, sku_name, source_database_id=None,
+                    recoverable_database_id=None, tags=None, no_wait=False):
+    # get the location of the workspace
+    workspace_client = cf_synapse_client_workspace_factory(cmd.cli_ctx)
+    workspace_object = workspace_client.get(resource_group_name, workspace_name)
+    location = workspace_object.location
+
+    max_size_bytes = None
+    sku_tier = None
     sku = Sku(tier=sku_tier, name=sku_name)
     sql_pool_info = SqlPool(sku=sku, location=location, max_size_bytes=max_size_bytes,
                             source_database_id=source_database_id, recoverable_database_id=recoverable_database_id,
-                            create_mode=create_mode, tags=tags)
+                            tags=tags)
 
     return sdk_no_wait(no_wait, client.create, resource_group_name, workspace_name, sql_pool_name, sql_pool_info)
+
+
+def update_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, sku_name=None, tags=None):
+    sku = Sku(name=sku_name)
+    sql_pool_patch_info = SqlPoolPatchInfo(sku=sku, tags=tags)
+    return client.update(resource_group_name, workspace_name, sql_pool_name, sql_pool_patch_info)
+
+
+def create_firewall_rule(cmd, client, resource_group_name, workspace_name, rule_name, start_ip_address, end_ip_address,
+                         no_wait=False):

```

Comment:
Are you going to support recovery from Backup and Point-in-time restore in this version?

Reply:
> Are you going to support recovery from Backup and Point-in-time restore in this version?

No. Let support it in next version.